stock_inventory_merge/models/wizard_stock_inventory_merge.py

```diff
@@ -51,6 +55,12 @@ def action_merge(self):
         for line in inventories.mapped("line_ids"):
             line.copy(default={"inventory_id": inventory.id})
 
+        if self.cancel_src:
+            # Note: `inventories.action_cancel_draft()` would remove the lines
```

Comment:
Hi. regarding that point, what do you think about a simple OCA module ``stock_inventory_draft_no_unlink`` that overwrite the function ``action_cancel_draft()`` like 

```
    def action_cancel_draft(self):
        self.mapped('move_ids')._action_cancel()
        self.write({
            'state': 'draft'
        })
```

And then, here, only call ``action_cancel_draft()``. I see that change quite generic and not linked to that specific module (``stock_inventory_merge``)

Reply:
Yes, that's right. But be aware that in my PR, we would like to have state `cancel`, not `draft`. But this could be easily solved.

I don't know the rationale of the Odoo core team for removing the lines when resetting the inventory to `draft`.
I mean, if the user wants to reset the inventory, they could always remove the lines manually afterwards.
IMHO [this method](https://github.com/OCA/OCB/blob/a082f6d9ae716561a40df61a5f77e1c089a97945/addons/stock/models/stock_inventory.py#L209-L214) violates the single-responsibility principle.

So yes, such a module "stock_inventory_draft_no_unlink" might be in order. Possibly providing also a separate method `action_cancel()` for model `stock.inventory`.
```
def action_cancel(self):
    self.mapped('move_ids')._action_cancel()
    self.write({
        'state': 'cancel'
    })

def action_cancel_draft(self):
    self.action_cancel()
    self.write({
        'state': 'draft'
    })
```

Also, I agree that this module "stock_inventory_merge" itself could also make a nice addition to OCA's `stock-logistics-warehouse`.

Reply:
Thinking about it again, the state `'draft'` normally occurs only when initially creating a new "stock.inventory". Such an inventory does not have any lines yet. Only when starting it, which means progressing into state `'confirm'` = "In Progess", the user can add lines.
So it makes indeed sense to remove the lines when resetting into state draft.
However, when merging inventories, the original ones should (IMHO) keep their lines. Consequently, this would then mean to not reset them back to state `'draft'`, but rather to keep them in `'confirm'`. Or set them to `'cancel'`, as per the suggested option.
Would you agree?

This would them mean
```python
def action_cancel(self):
    # cancel all moves
    self.mapped('move_ids')._action_cancel()
    self.write({
        'state': 'cancel',
    })

def action_cancel_draft(self):
    # first cancel
    self.action_cancel()
    # then set into 'draft', which implies inventory has no lines
    self.write({
        'line_ids': [(5,)],
        'state': 'draft'
    })
```

Regarding your second point: Yes, for confirmed, but not yet validated inventories, the `self.mapped('move_ids')._action_cancel()` is useless, since the inventory does not yet have stock moves.
However, the method may also be called from validated inventories, and in this case cancelling the stock moves is necessary.